draft-ietf-privacypass-architecture.md

```diff
@@ -544,6 +545,76 @@ and guidelines on how to deploy the protocol to minimize any privacy impacts.
 specified in {{issuer-role}} for managing Issuer public key data.
 
+## Information Flow {#flow}
+
+The end-to-end run of redemption and issuance protocol involves information
+flowing between Issuer, Origin, Client, and Attester. That information can
+have implications on the privacy goals that Privacy Pass aims to provide
+as outlined in {{privacy-and-trust}}. In this section, we describe the flow
+of information between each party. How this information affects the privacy
+goals in particular deployment models is further discussed in {{deployment}}.
+
+### Token Challenge Flow {#challenge-flow}
+
+To use Privacy Pass, Origins choose an Issuer from which they are willing to
+accept tokens. Origins then construct a token challenge using this specified
+Issuer and the redemption context it shares with the Client. This token
+challenge is then delivered to a Client. The token challenge conveys
+information about the Issuer and the redemption context, such as whether the
+Origin desires a per-Origin or cross-Origin token.
+
+### Attestation and Issuance Flow {#issue-flow}
+
+Clients use the issuance protocol to produce a token bound to a token
+challenge. In doing so, there are several ways in which the issuance protocol
+contributes information to the attestation context and issuance context.
+For example, a token request may contribute information to the attestation
+or issuance contexts as described below.
+
+- Issuance protocol. The type of issuance protocol can contribute information
+about the Issuer's capabilities to the attestation or issuance contexts.
+- Issuer identity or configuration. Information about the Issuer, such as its
+identity or the public key used to validate tokens it creates, can be revealed
+during issuance and contribute to the attestation or issuance contexts.
+- Attestation information. The issuance protocol can contribute information to
+the issuance context based on what attestation procedure the Issuer uses to
+trust a token request. In particular, a token request that is validated by
+a given Attester means that the Client which generated the token request must
+be capable of the completing the designated attestation procedure.
+- Origin information. The issuance protocol can contribute information about
+the Origin that challenged the Client in {{challenge-flow}}. In particular,
+a token request designated for a specific Issuer means that the resulting token
+is for an Origin that trusts the specified Issuer.
+
+Moreover, a token response may contribute information to the attestation
+or issuance contexts as described below.
+
+- Origin information. The issuance protocol can contribute information about
+the Origin in how it responds to a token request. For example, if an Issuer
+learns the Origin during issuance and is also configured to respond in some way
+on the basis of that information, that response can reveal information to the
+Attester.
```

Comment:
I'll repeat my previous complaint.  The Client interacting with the Issuer *via the Attester* is something that is assumed far too often in the document.  This needs to be conditional at a minimum.  Also, in those protocols that encapsulate the Client-Issuer exchange, the information that leaks is only based on traffic analysis.

Reply:
This is fair criticism. It's noted that this is an option in the overview section, but I don't think it's obvious enough. I'll try to clarify.